### DISCUSSION TITLE
Does the SDK handle all redirects that are returned to it?

### DISCUSSION THREAD

**Discussion by billygcode** [aws/aws-sdk-go-v2#1958] [Q&A]
Hello, I haven't been able to find a concrete answer to this questions.

When using the SDK, suppose you request PutObject or CreateBucket and it returns a redirect AWS. Would I have to set up my own redirecting service or will the SDK handle it?

**Comment by RanVaknin** [comment]
@18eddymedina ,
Not quite sure what you mean.

a `PutObject` request will return these 2 things, and **only** them: an `Err`, and a `putObjectOutput` struct.
If the server returns anything other than 200 **and** a valid serializeable JSON of that response shape, the SDK will panic and float that error.

**Reply by billygcode** [reply]
So I guess the question would be, would it ever return a 307 temporary redirect, or would the SDK function redirect immediately?
